### PR TITLE
Add protocol entities to PFB manifest (#3084)

### PR DESCRIPTION
**Pull request by danielsotirhos**
Connected issue: #3084

### Author

- [x] PR title references the connected issue
- [x] PR title matches<sup>1</sup> title of connected issue         <sub>or comment in PR explains why they're different</sub>
- [x] Title of at least one commit references connected issue
- [x] PR is connected to issue via Zenhub 
- [x] PR description links to connected issue

<sup>1</sup> when the issue title describes a problem, the PR title is `Fix: ` followed by the issue title   

### Author (reindex)

- [x] Added `r` tag to commit title                                 <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                                   <sub>or this PR does not require reindexing</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain                <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                        <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst          <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                                 <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                                   <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                          <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks           <sub>or this PR does not require additional tasks</sub>

## Author (hotfixes)

- [x] Added `h` tag to commit title and PR targets `prod`           <sub>or this PR does not include a temporary hotfix</sub>
- [x] Added `H` tag to commit title                                 <sub>or this PR does not include a permanent hotfix</sub>
- [x] Added `hotfix` label to PR                                    <sub>or this PR does not include a hotfix</sub>
- [x] Reverted the temporary hotfix connected to the issue          <sub>or there is no temporary hotfix for the connected issue on the `prod` branch</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                                <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                                 <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                                      <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment         <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Labeled connected issue as `demo` or `no demo`
- [x] Commented on connected issue about demo expectations          <sub>or labelled connected issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear                      <sub>or connected issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label             <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                                       <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                                  <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                             <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved connected issue to Merged column
- [x] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                         <sub>or PR description does not contain an announcement</sub>
- [ ] Shortened the PR chain                                        <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate               <sub>or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                                 <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab
- [ ] Moved connected issue to `prod` or `Merged prod`              <sub>or this PR does not represent a promotion</sub>

### Operator (reindex) 

- [ ] Started reindex in target deployment                          <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures                     <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment                      <sub>or this PR does not require reindexing</sub>
- [ ] Filed backport PR                                             <sub>or this PR does not represent a hotfix</sub>

### Operator

- [ ] Unassigned PR

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Review by noah-aviel-dove — APPROVED** [review]
Looks straightforward, approved :+1:

File: test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json [hunk 1]
```diff
@@ -1675,6 +1675,9 @@
                 "imaging_protocols": [],
                 "library_preparation_protocols": [
                     {
+                        "document_id": [
```
Comment (hannes-ucsc) [inline test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json:1678]: I don't think this should be a list in contributions. Each dict in the outer list represents a single protocol and one protocol can only have one ID. Correct me if I am wrong but the other inner entities in contributions don't have lists for document_id either.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json (hunk 1 above)
Comment (danielsotirhos) [reply test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json:1678]: In an aggregate document all inner entities (except `files` entities in a `bundles` or `files` aggregate document) either don't include the `document_id` field, or it is a list.
Examples:
- [Sequencing process inner entity in bundles aggregate ](https://github.com/DataBiosphere/azul/blob/19fdbcf8a5ffc6b8e22d2077a14f1f408003510c/test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json#L1506)
- [Specimens inner entity in bundles aggregate](https://github.com/DataBiosphere/azul/blob/19fdbcf8a5ffc6b8e22d2077a14f1f408003510c/test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json#L1514)
- [Donor inner entity in files aggregate](https://github.com/DataBiosphere/azul/blob/19fdbcf8a5ffc6b8e22d2077a14f1f408003510c/test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json#L2234)
- [Cell suspension inner entity in samples aggregate](https://github.com/DataBiosphere/azul/blob/19fdbcf8a5ffc6b8e22d2077a14f1f408003510c/test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json#L2565)

Comment (danielsotirhos) [reply test/indexer/data/aaa96233-bf27-44c7-82df-b4dc15ad4d9d.2018-11-02T113344.698028Z.results.json]: @hannes-ucsc : "Apparently I mistook this occurrence as being part of the contribution. Daniel and I went through this and did not find any such occurrence. The only occurrences of arrays in `document_id` are in aggregates."

**Review by hannes-ucsc — APPROVED** [review]
I'm a little worried about the < 100 protocol `document_id` values this is going to add to project aggregates. We need to keep an eye on this.

**Pull request merged**